### ISSUE TITLE
FTBFS (32-bit): overflow in implicit constant conversion

### ISSUE THREAD

**Issue by andrewpollock** [grpc/grpc#2950]
This is a reproduction of the bug report at https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=795766

In an ideal world, correpondence including 795766-forwarded@bugs.debian.org would keep our bug tracking system in the loop.

Builds of grpc for 32-bit architectures (i386 and powerpc so far) have                                                  
been failing:                                                                                                           

```                                                                                                                        
  [C]       Compiling test/core/iomgr/tcp_server_posix_test.c                                                           
  In file included from test/core/iomgr/tcp_server_posix_test.c:39:0:                                                   
  test/core/iomgr/tcp_server_posix_test.c: In function 'test_connect':                                                  
  ./test/core/util/test_config.h:52:3: error: overflow in implicit constant conversion [-Werror=overflow]               
     (GRPC_TEST_SLOWDOWN_BUILD_FACTOR * GRPC_TEST_SLOWDOWN_MACHINE_FACTOR)                                              
     ^                                                                                                                  
  ./test/core/util/test_config.h:56:37: note: in expansion of macro 'GRPC_TEST_SLOWDOWN_FACTOR'                         
                  gpr_time_from_micros(GRPC_TEST_SLOWDOWN_FACTOR * 1e6 * (x)))                                          
                                       ^                                                                                
  test/core/iomgr/tcp_server_posix_test.c:128:16: note: in expansion of macro 'GRPC_TIMEOUT_SECONDS_TO_DEADLINE'        
       deadline = GRPC_TIMEOUT_SECONDS_TO_DEADLINE(4000);                                                               
                  ^                                                                                                     
  cc1: all warnings being treated as errors                                                                             
  make[2]: *** [objs/opt/test/core/iomgr/tcp_server_posix_test.o] Error 1                                               
  Makefile:2847: recipe for target 'objs/opt/test/core/iomgr/tcp_server_posix_test.o' failed                            
  make[2]: Leaving directory '/«PKGBUILDDIR»'                                                                           
  make[1]: *** [override_dh_auto_test] Error 2                                                                          
  make: *** [build-arch] Error 2                                                                                        
  debian/rules:22: recipe for target 'override_dh_auto_test' failed                                                     
  make[1]: Leaving directory '/«PKGBUILDDIR»'                                                                           
  debian/rules:29: recipe for target 'build-arch' failed                                                                
  dpkg-buildpackage: error: debian/rules build-arch gave error exit status 2
```

**Comment by ctiller** [comment]
Hey Anthony,

It looks like a test-only compilation failure. I'm getting a test
environment set up here, and hope to make the fixes shortly.

Craig

**Comment by ctiller** [comment]
This is fixed at head. Still needs to be backported to release 0.10.1

**Comment by tbetbetbe** [comment]
This was fixed in #2985, and include in release 0.10.2
